PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/MoveLearnMethodLocations/MoveLearnMethodLocationSheetRowParserTest.cs

```diff
@@ -0,0 +1,154 @@
+﻿using FluentAssertions;
+using PokeOneWeb.DataSync.GoogleSpreadsheet.Import.Impl;
+using PokeOneWeb.DataSync.GoogleSpreadsheet.Import.Impl.Sheets.MoveLearnMethodLocations;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace PokeOneWeb.DataSync.Tests.GoogleSpreadsheet.Import.Impl.Sheets.MoveLearnMethodLocations
+{
+    public class MoveLearnMethodLocationSheetRowParserTest
+    {
+        [Fact]
+        public void ReadRow_WithMinimalValidValues_ShouldParse()
+        {
+            // Arrange
+            var parser = new MoveLearnMethodLocationSheetRowParser();
+
+            var moveLearnMethodName = "Move Learn Method Name";
+            var tutorType = "Tutor Type";
+            var npcName = "NPC Name";
+            var locationName = "Location Name";
+
+            var values = new List<object>
+            {
+                moveLearnMethodName, tutorType, npcName, locationName
+            };
+
+            var expected = new MoveLearnMethodLocationSheetDto
+            {
+                MoveLearnMethodName = moveLearnMethodName,
+                TutorType = tutorType,
+                NpcName = npcName,
+                LocationName = locationName
+            };
+
+            // Act
+            var result = parser.ReadRow(values);
+
+            // Assert
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void ReadRow_WithAllValidValues_ShouldParse()
+        {
+            // Arrange
+            var parser = new MoveLearnMethodLocationSheetRowParser();
+
+            var moveLearnMethodName = "Move Learn Method Name";
```

Comment:
Maybe a look at AutoFixture could be something to decrease the amount of time to invest into setup the test data. But I am also quite lazy ;).

Reply:
Yes quite a lot of (also sometimes duplicated) lines of code are used to set up test data and I have been thinking about changing something about it mutliple time. But I prefer more code over complicated and obfuscating solutions. With the current solution it is always abundantly clear what the exact test setup is, and it is always possible to change the setup without influencing other tests. But may be a point to discuss.